NewForce-Cohort-7/tabloidmvc-mollasses-gang
# Gr ticket8: Added Features to Edit/Delete a Post

# Description
The functionality to delete a post has been added(Ticket#7):https://trello.com/c/ChCiygB6/7-7-delete-a-post
The functionality to edit a post has been added(Ticket#8):  https://trello.com/c/V8N43rvX/8-8-edit-a-post

Delete:
Given a user is logged in the app & selects the Post link in the navbar, they should be directed to an  'Index' view page that lists all the posts. Given a user clicks on the trash can icon of the post they wish to delete, they should be directed to a form in which they can delete the selected post. Given the user wishes to confirm the delete, when the user clicks the Delete button, they should be redirected to the Category list page and will no longer see the deleted category name in the list.Given the user has decided not to delete the selected post, when they click the Cancel button, the user should be redirected back to the list page

Edit:
Given a user is logged in the app & selects the Post link in the navbar, they should be directed to an  'Index' view page that lists all the posts. Given a user clicks on the pencil icon of the post they wish to edit, they should be directed to a form in which they can edit the selected post. Given the user is done with their edits, when the user clicks the save button, they should be redirected to the Category list page and will see their selected post with the recent edits in the list.Given the user has decided not to edit the selected post, when they click the Cancel button, the user should be redirected back to the list page


# Additional Information
 -For the time being  all users are admin users. 


# Type of change:
*Two new feature (non-breaking change which adds functionality)
*Created the following files and methods:
         

                  1. Delete method added in Repositories>CategoryRepository.cs
                  2. Delete method added in Repositories>ICategoryRespository.cs
                  3. Delete (GET) and Delete(POST) methods added in CategoryController.cs
                  4. File for Delete in Views>Category>Delete.cshtml
                  5. Edit method added in Repositories>CategoryRepository.cs
                  6. Edit method added in Repositories>ICategoryRespository.cs
                  7. Edit(GET) and Edit(POST) methods added in CategoryController.cs
                  8. File for Edit in Views>Category>Edit.cshtml



# Testing Instructions
- Reference [How to test a teammate's PR](https://gist.github.com/jordan-castelloe/96f76d5bd4cfbd2181084af674c2a1af#when-you-need-to-test-a-teammates-pr)
- git checkout  grTicket8
- If you don't see the most recent version of my files, git pull origin grTicket8
- Add all seed data by adding a new query and copying the code from the links below
- Use this to fill out tables and columns:
https://github.com/NewForce-Cohort-7/tabloidmvc-mollasses-gang/blob/main/SQL/01_TabloidMVC_Create_DB.sql
- Use this to seed the database:
https://github.com/NewForce-Cohort-7/tabloidmvc-mollasses-gang/blob/main/SQL/02_TabloidMVC_Seed_Data.sql
- To execute your query just press Ctrl+Shift+e inside of your query.
- Run the program by clicking the Green Arrow next to TabloidMVC at the top.
- BEFORE TESTING Run another query using this code INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
VALUES ('Sample Title', 'Lorem ipsum dolor sit amet.', 'https://example.com/image.jpg', '2023-07-05 10:00:00', '2023-07-06', 1, 1, 1); (Use this to test Edit and Delete Function!)') and execute by pressing Ctrl+Shift+e.
- In the  browser, please go to Login on the nav bar 
- Log in with this email: [admin@example.com](mailto:admin@example.com). This should direct you back to the home page.
- Click on Post  in the  NavBar
- View the list of available Posts

# To Edit:
- Click on the pencil icon of the 'Sample Title'post you added in order to edit
- User will be directed to a form in which they can edit the post
- User makes appropriate edits and clicks the save button
- User will be redirected to the post list page and will see selected post with recent edits in the list
- Given the User has decided not to edit the selected post
- When User clicks the Cancel button, the user should be redirected to the index view page where they will see a list of posts
- If User uses the url to select a post that doesn’t exist in the database i.e:https://localhost:7131/Category/Edit/20
- The error message:"Ooops. This post does not seem to exist!" will be displayed 

# To Delete:
- Click on the trash can icon of the 'Sample Title 'post you created to delete
- User will be directed to a form in which they can delete the post
- To confirm deletion, Click the Delete button
- User will be redirected to the Category list page and will no longer see the deleted category
- Given the User has decided not to delete the selected post
- When User clicks the Cancel button, the user should be redirected to the index view page where they will see a list of categories
- If User uses the url to select a category that doesn’t exist in the database i.e:https://localhost:7131/Category/Delete/20
- The error message:"Ooops. This post does not seem to exist!" will be displayed 
- Locate Logout on the Navbar & click this to log out.


# Checklist:


- [ x] - My code follows the style guidelines of this project
- [x ] - I have performed a self-review of my code
- [x ] - I have commented my code, particularly in hard-to-understand areas
- [x ] - My changes generate no new warnings or errors
- [x ] - I have added test instructions that prove that my feature works
